test_programs/compile_success_empty/comptime_typed_expr/src/main.nr

```diff
@@ -0,0 +1,7 @@
+fn main() {
+    comptime
+    {
+        let expr = quote { [1, 3] }.as_expr().unwrap().resolve(Option::none());
```

Comment:
In another PR we should add

```rs
fn as_typed_expr(q: Quoted, scope: Option<FunctionDefinition>) -> Option<TypedExpr> {
    q.as_expr().map(|e: Expr| e.resolve(scope))
} 
```
One issue is that passing None would resolve within `as_typed_expr` instead of `main` though. You wouldn't be able to reference local variables even if you passed main explicitly I think.